editor/src/components/canvas/canvas-strategies/reparent-strategy-helpers.ts

```diff
@@ -287,68 +550,295 @@ export function applyFlexReparent(
       ) {
         const target = filteredSelectedElements[0]
         const newParent = reparentResult.newParent
+        const newParentADescendantOfCurrentParent = EP.isDescendantOfOrEqualTo(
+          newParent,
+          EP.parentPath(target),
+        )
         // Reparent the element.
         const { commands: reparentCommands, newPath } = getReparentOutcome(
           canvasState.builtInDependencies,
           canvasState.projectContents,
           canvasState.nodeModules,
           canvasState.openFile,
           target,
-          reparentResult.newParent,
+          newParent,
+          'on-complete',
         )
 
         // Strip the `position`, positional and dimension properties.
-        const commandToRemoveProperties =
-          stripAbsoluteProperties === 'strip-absolute-props'
-            ? [deleteProperties('always', newPath, absolutePropsToRemove)]
-            : []
+        const commandToRemoveProperties = stripAbsoluteProperties
+          ? [
+              deleteProperties('on-complete', newPath, propertiesToRemove),
+              setProperty('on-complete', newPath, PP.create(['style', 'position']), 'relative'), // SPIKE TODO only insert position: relative if there was a position nonstatic prop before
+            ]
+          : []
 
         const commandsBeforeReorder = [
           ...reparentCommands,
-          updateSelectedViews('always', [newPath]),
+          updateSelectedViews('on-complete', [newPath]),
         ]
 
         const commandsAfterReorder = [
           ...commandToRemoveProperties,
-          setElementsToRerenderCommand([newPath]),
-          updateHighlightedViews('mid-interaction', []),
-          setCursorCommand('mid-interaction', CSSCursor.Move),
+          setElementsToRerenderCommand([target, newPath]),
+          // updateHighlightedViews('on-complete', []), // TODO WELP
```

Comment:
Does this not work anymore?

Reply:
omg it was so long ago. I wish I wrote something more useful here :D I'll try to uncomment it and see what broke